tests/test_clean_collections.py

```diff
@@ -0,0 +1,204 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import os
+import shutil
+import tempfile
+import unittest
+
+import lsst.utils.tests
+import yaml
+from lsst.ctrl.oods.fileIngester import FileIngester
+from lsst.daf.butler import Butler, CollectionType
+from lsst.daf.butler.tests import MetricsExample, addDataIdValue, addDatasetType, registerMetricsExample
+
+
+class CleanCollectionsTestCase(unittest.IsolatedAsyncioTestCase):
+
+    def setUp(self):
+        """set up test info, including config dict, and populate
+        Butler repo with test data
+        """
+
+        config_name = "clean_collections.yaml"
+
+        # create a path to the configuration file
+
+        testdir = os.path.abspath(os.path.dirname(__file__))
+        configFile = os.path.join(testdir, "etc", config_name)
+
+        # load the YAML configuration
+
+        with open(configFile, "r") as f:
+            self.config = yaml.safe_load(f)
+
+        # extract parts of the ingester configuration
+        # and alter the image staging directory to point
+        # at the temporary directories created for his test
+
+        ingesterConfig = self.config["ingester"]
+        self.imageDir = tempfile.mkdtemp()
+        ingesterConfig["imageStagingDirectory"] = self.imageDir
+
+        self.badDir = tempfile.mkdtemp()
+        butlerConfig = ingesterConfig["butlers"][0]["butler"]
+        butlerConfig["badFileDirectory"] = self.badDir
+        self.stagingDir = tempfile.mkdtemp()
+        butlerConfig["stagingDirectory"] = self.stagingDir
+
+        self.repoDir = tempfile.mkdtemp()
+        Butler.makeRepo(self.repoDir)
+
+        butlerConfig["repoDirectory"] = self.repoDir
+
+        self.clean_collections = butlerConfig["cleanCollections"]
+
+        # Define the run collection
+        run_a = "collection_a"
+        run_b = "collection_b"
+        self.collections = [run_a, run_b]
+
+        # Initialize the Butler
+        opts = dict(writeable=True)
+        self.butler = Butler(self.repoDir, **opts)
+
+        self.butler.registry.registerCollection("collection_a", CollectionType.RUN)
+        self.butler.registry.registerCollection("collection_b", CollectionType.RUN)
```

Comment:
You defined auxiliary variables with collections' names (`run_a` and `run_b`) earlier. Can we (re)use them in these two lines?

Reply:
I was refactoring things and missed those.  Fixed.